cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/template/etl/batch/BatchTransformContext.java

```diff
@@ -42,6 +42,11 @@ public Metrics getMetrics() {
   }
 
   @Override
+  public <T> T newPluginInstance(String pluginId) throws InstantiationException {
+    return context.newPluginInstance(pluginId);
```

Comment:
this is a weird transition thing until we remove templates, but we have to `try { return context.newPluginInstance(pluginId); } catch (UnsupportedOperationException e) { return context.newInstance(pluginId); }` to support both templates and apps